cmd/auto-updater/main.go

```diff
@@ -167,37 +162,50 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 		return nil
 	}
 
-	err = loop.Run(ctx, func(ctx context.Context) (err error) {
+	loopInterval, err := time.ParseDuration(interval)
+	if err != nil {
+		return errs.New("unable to parse interval parameter: %v", err)
+	}
+
+	loopFunc := func(ctx context.Context) (err error) {
 		if err := update(ctx); err != nil {
 			// don't finish loop in case of error just wait for another execution
 			log.Println(err)
 		}
 		return nil
-	})
+	}
+
+	if loopInterval == 0 {
+		err = loopFunc(ctx)
+	} else {
+		loop := sync2.NewCycle(loopInterval)
+		err = loop.Run(ctx, loopFunc)
+	}
 	if err != context.Canceled {
 		return err
 	}
 	return nil
 }
 
-func binaryVersion(location string) (version.SemVer, error) {
+func binaryVersion(location string) (semver.Version, error) {
 	out, err := exec.Command(location, "version").Output()
 	if err != nil {
-		return version.SemVer{}, err
+		return semver.Version{}, err
 	}
 
 	scanner := bufio.NewScanner(bytes.NewReader(out))
 	for scanner.Scan() {
 		line := scanner.Text()
 		prefix := "Version: "
 		if strings.HasPrefix(line, prefix) {
-			return version.NewSemVer(line[len(prefix):])
+			// use version string without 'v'
+			return semver.Make(line[len(prefix)+1:])
```

Comment:
How are we sure that in this case the version starts with `v`? Should we have an additional `strings.HasPrefix(line, "v")` check?

Reply:
yeah, additional check makes sense. I will add it.